syntax = "proto3";

package training;

message TrainingMetrics {
  string session_id = 1;
  int32 step = 2;
  int32 episode = 3;
  float loss = 4;
  float reward = 5;
  optional float q_value = 6;
  int32 episode_length = 7;
}

// Request message for streaming training metrics
message TrainingMetricsRequest {
  string session_id = 1;
}

service TrainingService {
  // Server-side streaming RPC to send training metrics updates
  rpc StreamTrainingMetrics(TrainingMetricsRequest) returns (stream TrainingMetrics);
}